python/pyspark/sql/dataframe.py

```diff
@@ -78,6 +78,7 @@ def __init__(self, jdf, sql_ctx):
         self.is_cached = False
         self._schema = None  # initialized lazily
         self._lazy_rdd = None
+        self._support_repr_html = False
```

Comment:
Shall we explain why we need this (as talked in https://github.com/apache/spark/pull/21370#discussion_r191591799)? It took me a while to understand too.

Reply:
Got it, more comments in 7f43a8b.